### DISCUSSION TITLE
Unkown Error after HA reinstall

### DISCUSSION THREAD

**Discussion by JeremyJvR** [rainepretorius/olarm-ha-integration#103] [General]
Good day all,

I had to reinstall my HA after the supervisor didn't want to update. After this my Olarm integration is not working. Upon reconfiguration, I just get a "Unknown Error"

![image](https://github.com/rainepretorius/olarm-ha-integration/assets/86603135/9fe008da-2ee4-4ed0-baf0-007d644f731d)

Any ideas?

**Comment by JeremyJvR** [comment]
This morning I am getting this.

`ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform olarm_sensors.alarm_control_panel: Platform not found (cannot import name 'DeviceInfo' from 'homeassistant.helpers.device_registry' (/usr/src/homeassistant/homeassistant/helpers/device_registry.py)).`

**Reply by D1ac1d3** [reply]
@JeremyJvR Ever got this sorted? I upgraded my Olarm 2g to a Olarm 4g, and had to reconfigure. Now I am getting the same error.
![Screenshot 2024-07-28 101219](https://github.com/user-attachments/assets/f711f0b3-3d7b-45e4-a21c-a584d2fc80b4)

**Reply by boereworswizard** [reply]
> @JeremyJvR Ever got this sorted? I upgraded my Olarm 2g to a Olarm 4g, and had to reconfigure. Now I am getting the same error. ![Screenshot 2024-07-28 101219](https://private-user-images.githubusercontent.com/176796657/352798527-f711f0b3-3d7b-45e4-a21c-a584d2fc80b4.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5MjcxMTMsIm5iZiI6MTcyOTkyNjgxMywicGF0aCI6Ii8xNzY3OTY2NTcvMzUyNzk4NTI3LWY3MTFmMGIzLTNkN2ItNDVlNC1hMjFjLWE1ODRkMmZjODBiNC5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyNlQwNzEzMzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MDRkYmMzZDkwY2Y5YTZiNTUzOWY1OWU4NjRiZWZkMGI4YzhiM2U5MTk1YjRkYTRlYzYyNTcxN2RhYmUwYTVkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.sbcSzv4zxICizn2m8J4otKjofaYWfBGtThtjxTo6rbM)

Did you find any fix?

**Comment by JeremyJvR** [comment]
Hi Yes,

On the Olarm site you have to select your device when creating the API token. That worked for me.

**Reply by D1ac1d3** [reply]
Hi, thanks for the reply. What do you mean by "Select Your Device". I don't see that option on the API access page.
![image](https://github.com/user-attachments/assets/7e859213-c789-4619-96d4-6c995985e9ea)

**Reply by JeremyJvR** [reply]
You can find it under "Devices"

![image](https://github.com/user-attachments/assets/8213e815-9cac-4b69-b327-3c46eca6e6b2)

**Comment by D1ac1d3** [comment]
When I click on the Device, all it does is take me to the device setup page, with nothing I can see relating to API access.

![image](https://github.com/user-attachments/assets/a0926e0b-0880-4b81-b956-e46624665743)

**Comment by JeremyJvR** [comment]
mmmmm, that worked for when I re-registered the device and then generated the API. I then went back and configured in HA. I must add that the period between I loaded the issue and reconfigured was about a month with another update in between

**Reply by D1ac1d3** [reply]
I am going to try and remove the device on Olarm and add it again. I hope this won't affect my subscription.

**Reply by D1ac1d3** [reply]
That also did not work, I will create a bug report.

**Reply by JeremyJvR** [reply]
I am having exactly the same issue again after updating HA to the latest version.

**Reply by D1ac1d3** [reply]
I solved my problem. I deleted the whole olarm_sensors folder underneath the custom_components folder. I then downloaded a fresh copy of the olarm_sensors from the github page, and uploaded that to the custom_components folder. I generated a new api key on the olarm api page, did a HA server restart and added the Olarm sensors integration again from the devices and services menu. The olarm sensors integration is not available on HACS as the documentation states. You must do a manual install/reinstall.

**Reply by RolandSwe** [reply]
I just did a brand new install, i.e. my first time installing the integration.  It has given me the same error.  I’m new to HACS so don’t want to break my setup.  Anyone got any other ideas for fixes?

**Comment by RolandSwe** [comment]
I just uninstalled the integration and reinstalled it.  Still giving the same error, not sure what else to try?

**Reply by D1ac1d3** [reply]
Do you have studio code server installed? How are you installing the integration. Cause according to me, it is no longer available on HACS. You must do a manual install, and the easiest way is to do that through Studio Code Server.

**Reply by RolandSwe** [reply]
I do have it installed, and use that to edit my yaml file.  I found it on HACS when I searched for "olarm-ha-integration". 

How do i do a manual install?

**Reply by RolandSwe** [reply]
Can't seem to find instructions.

**Reply by D1ac1d3** [reply]
Delete the whole olarm_sensors folder under you customizations folder. Download the latest version from the olarm Github page, and upload that to the customizations folder. Restart HA, and check again.
 
![image](https://github.com/user-attachments/assets/f917e0cb-30ec-4402-b584-f47e97d70a6a)

![image](https://github.com/user-attachments/assets/447d1663-590c-4b4d-be88-c007605ece72)

**Reply by RolandSwe** [reply]
Thank you! This all made sense, I removed it from the custom folder.   I even went and deleted it from HACS(It was still there after i deleted it from the folder).  I restarted, added the folder back to the custom folder, restarted again.   Got a new API, tried to add Olarm under add devices but got the same issue.  
I have two different large panel setups on the smae account, so I made the api call interval at 20secs, then tried again.  But no luck.  I'm going to keep trying.   When I do the API call through my browser it is working, i can see all the zones and alarm statuses on both panels.

Could it be because I have the older gen Olarm that is only on cellular?

**Reply by D1ac1d3** [reply]
Yeah, not sure if the model of olarm device is an issue, these are the steps I followed to get mine going again.

**Reply by D1ac1d3** [reply]
A full explanation of what I did to get mine going again is on the bug report here, last post:

https://github.com/rainepretorius/olarm-ha-integration/issues/113

**Reply by RolandSwe** [reply]
So i managed after multiple attempts to get my olarm integrated....but....it now has no entities.  Am i being a COMPLETE noob here?

**Reply by RolandSwe** [reply]
Also @D1ac1d3 thank you for you support on this, really appreciat it
.

**Reply by D1ac1d3** [reply]
> So i managed after multiple attempts to get my olarm integrated....but....it now has no entities. Am i being a COMPLETE noob here?

 Have you added the Olarm sensors integration again from the devices and services menu?

**Reply by JeremyJvR** [reply]
Same on my side, did a complete reinstall. No entities, such a pain as my automations are not running.

**Reply by JeremyJvR** [reply]
Got it working by restarting HA and reloading the integration by selecting the "ellipses" next to "Configure" and selecting "Reload"

**Reply by boereworswizard** [reply]
> Got it working by restarting HA and reloading the integration by selecting the "ellipses" next to "Configure" and selecting "Reload"

Still working for you?
